yarn-project/acir-simulator/src/client/private_execution.ts

```diff
@@ -43,7 +40,7 @@ export class PrivateFunctionExecution {
     private argsHash: Fr,
     private callContext: CallContext,
     private curve: Grumpkin,
-    private orderings: Orderings = { publicCall: 0 },
+    private sideEffectCounter: number = 0,
```

Comment:
I think a single global side-effect counter is sufficient and simpler. In fact certain different events/actions will need to have their order validated relative to one another (reads & updates, commitments & reads & nullifiers, etc), so having a single counter for all such events may make more sense.

Reply:
Sounds good. I had considered one counter per "thing" because it allows us to detect gaps in ordering when validating that these are sound in the kernel circuit. But I'm not sure it'll even be needed, and if it is, we can always change it later.